filament/backend/src/metal/MetalDriver.mm

```diff
@@ -1388,8 +1388,11 @@
         }
         const auto* metalSamplerGroup = mContext->samplerBindings[samplerGroupIdx];
         if (!metalSamplerGroup) {
-            utils::slog.w << "Program has non-empty samplerGroup (index " << samplerGroupIdx <<
-                    ") but has not bound any samplers." << utils::io::endl;
+#ifndef NDEBUG
```

Comment:
I believe we don't need this even in DEBUG.

Reply:
Agreed.  Changed the PR accordingly.